src/index.js

```diff
@@ -15,3 +25,55 @@ module.exports = async function ({ github, context, inputs, packageVersion }) {
 
   logError('Unsupported event')
 }
+
+async function bumpVersion({ inputs }) {
+  const newVersion =
+    inputs.semver === autoInput
+      ? await getAutoBumpedVersion(inputs['base-tag'])
+      : inputs.semver
+
+  const preReleasePrefix = inputs['prerelease-prefix'] || ''
+
+  const run = runSpawn()
+  await run('npm', [
+    'version',
+    '--no-git-tag-version',
+    `--preid=${preReleasePrefix}`,
+    newVersion,
+  ])
+  return await run('npm', ['pkg', 'get', 'version'])
+}
+
+async function getAutoBumpedVersion(baseTag = null) {
+  try {
+    const run = runSpawn()
+    await run('git', ['fetch', '--unshallow']) // by default optic does a shallow clone so we need to do this to get full commit history
+    await run('git', ['fetch', '--tags'])
+
+    let latestTag = null
+    if (!baseTag) {
+      const allTags = await run('git', ['tag', '--sort=-creatordate'])
+      const tags = allTags.split('\n')
+      latestTag = tags[0] || null
+    }
+
+    const tag = baseTag || latestTag
+
+    logInfo(`Using ${tag} as base release tag for version bump`)
+
+    const { releaseType = 'patch' } = await conventionalRecommendedBumpAsync({
+      baseTag: tag,
+      config: conventionalCommitsConfig,
+    })
+    logInfo(`Auto generated release type is ${JSON.stringify(releaseType)}`)
+    return releaseType
+  } catch (error) {
+    core.setFailed(error.message)
+    throw error
```

Comment:
this error handling seems unnecessary. let's make sure error handling is done properly in the root of the action, which means that calling setFailed here and rethrowing the error shouldn't be needed

Reply:
This is one is called directly from `action.yaml`. I too think we can remove this error handling since if this function throws the action should fail and report error message automatically.